reader.go

```diff
@@ -199,35 +200,45 @@ func (r *Reader) ReadDouble() float64 {
 
 // ReadBytes reads Bytes from the Reader.
 func (r *Reader) ReadBytes() []byte {
-	size := r.ReadLong()
-	if size < 0 {
-		r.ReportError("ReadBytes", "invalid bytes length")
-		return nil
-	}
-
-	buf := make([]byte, size)
-	r.Read(buf)
-	return buf
+	return r.readBytes("bytes")
 }
 
 // ReadString reads a String from the Reader.
 func (r *Reader) ReadString() string {
+	b := r.readBytes("string")
+	if len(b) == 0 {
+		return ""
+	}
+
+	return *(*string)(unsafe.Pointer(&b))
+}
+
+func (r *Reader) readBytes(typ string) []byte {
 	size := int(r.ReadLong())
 	if size < 0 {
-		r.ReportError("ReadString", "invalid string length")
-		return ""
+		r.ReportError("ReadString", "invalid "+typ+" length")
```

Comment:
The only point of the `typ` is error reporting? 🤔 In the case of the error here, does the type really matter? the error is that the size is `<0` and has nothing to do with the type, does it?